### PR TITLE
CI: Check that blog builds without warnings

### LINKED ISSUE #25: Check for mkdoc warnings
Can make mkdocs stricter to fail in CI when there are warnings.

### PR DESCRIPTION
**Pull request by alexdewar**
Add a workflow to run MkDocs in strict mode to check for build warnings and errors.

Closes #25.

### CODE REVIEW

**Review by SaranjeetKaur — APPROVED** [review]
nice one!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: CI checks
+
+on:
+  push:
+    branches: [main]
```
Comment (cc-a) [inline .github/workflows/ci.yml:5]: Won't this be inherently covered by the publish workflow?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request:
+  release:
```
Comment (cc-a) [inline .github/workflows/ci.yml:7]: I don't think we're using releases.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          poetry-version: 1.8.3
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: poetry
+      - name: Install dependencies
+        run: poetry install
+      - name: Check that documentation builds cleanly with MkDocs
+        run: poetry run mkdocs build -s
```
Comment (cc-a) [inline .github/workflows/ci.yml:28]: Super picky but I think it's nice to use the long form of flags in this sort of context for readability.

**Review by cc-a — CHANGES_REQUESTED** [review]
Nice. A few suggestions.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (alexdewar) [reply .github/workflows/ci.yml:5]: Kiiiind of. We're not building the docs in strict mode for the publish workflow though, which means warnings could creep in.

How about using strict mode for the publish workflow?

Comment (cc-a) [reply .github/workflows/ci.yml:5]: That makes sense. At the moment having this workflow run on push wouldn't actually stop the publication from happening anyway so best to mae the strictness consistent.

**Review by cc-a — APPROVED** [review]
Great. Good stuff.

**Pull request merged**